src/archive.rs

```diff
@@ -241,10 +243,12 @@ impl<'a> EntriesFields<'a> {
         }
 
         let file_pos = self.next;
-        let size = header.entry_size()?;
-
+        let mut size = header.entry_size()?;
+        if (size == 0) && (pax_size.is_some()) {
+            size = pax_size.unwrap();
+        }
         let ret = EntryFields {
-            size: size,
+            size: header.entry_size()?,
```

Comment:
Could this field be removed entirely? That way there's no need to keep it in sync, and users can use the `size()` method